### PR TITLE
[utils.sh] Split `addOrEditKeyValuePair` to do `addKey` in a separate function

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Having a function with a variable number of arguments in this case was too complicated for something that is only really used once

### CODE REVIEW

**Review by yubiuser — CHANGES_REQUESTED** [review]
If splitting to add `addkey` the wrappers in `webpage.sh` need adjustment.

File: advanced/Scripts/webpage.sh [hunk 1]
```diff
@@ -147,67 +147,67 @@ SetWebPassword() {
...
+    removeKey "${dnsmasqconfig}" "server"
 
     COUNTER=1
     while true ; do
         var=PIHOLE_DNS_${COUNTER}
         if [ -z "${!var}" ]; then
             break;
         fi
-        add_dnsmasq_setting "server" "${!var}"
+        addOrEditKeyValPair "${dnsmasqconfig}" "server" "${!var}"
```
Comment (yubiuser) [inline advanced/Scripts/webpage.sh:158]: I think this is not going to work. If I understand the whole part right, it is reading `PIHOLE_DNS_X` from `setupVars` and adding it to `01-pi-hole.conf`. 
`addOrEditKeyValPair` is preventing adding multiple `server` lines, as it changes them always. You need

```suggestion
        addKey "${dnsmasqconfig}" "server=${!var}"
```

Comment (PromoFaux) [reply advanced/Scripts/webpage.sh:158]: Ah yes. That is a nice point. Currently mobile, so difficult to look - but do existing values get cleared out first, before we loop through PIHOLE_DNS_ values?

Comment (yubiuser) [reply advanced/Scripts/webpage.sh:158]: Yes.

https://github.com/pi-hole/pi-hole/blob/23e6fa1ec56e7e24054d359ef5da0114e2f9b77f/advanced/Scripts/webpage.sh#L147-L154

File: test/test_any_utils.py [hunk 2]
```diff
@@ -6,8 +6,8 @@ def test_key_val_replacement_works(host):
     addOrEditKeyValPair "./testoutput" "KEY_TWO" "value2"
     addOrEditKeyValPair "./testoutput" "KEY_ONE" "value3"
     addOrEditKeyValPair "./testoutput" "KEY_FOUR" "value4"
-    addOrEditKeyValPair "./testoutput" "KEY_FIVE_NO_VALUE"
-    addOrEditKeyValPair "./testoutput" "KEY_FIVE_NO_VALUE"
+    addKey "./testoutput" "KEY_FIVE_NO_VALUE"
+    addKey "./testoutput" "KEY_FIVE_NO_VALUE"
```
Comment (yubiuser) [inline test/test_any_utils.py:10]: I'm not sure what you want to do here. Why do you add it twice?

It this is what you really want to check for, adjust the check in L15 to

```
expected_stdout = 'KEY_ONE=value3\nKEY_TWO=value2\nKEY_FOUR=value4\nKEY_FIVE_NO_VALUE\nKEY_FIVE_NO_VALUE\n'
``

Comment (PromoFaux) [reply test/test_any_utils.py:10]: Ah, no. Thinking on it... AddKey is missing some logic to check for dupes before adding

Comment (yubiuser) [reply test/test_any_utils.py:10]: Right. Good Point!

Comment (yubiuser) [reply test/test_any_utils.py:10]: I was wrong... we do have a check on L54
````
if ! grep -q "^${key}" "${file}"; then
`````
But still I think it makes not much sense  to add it twice.

Comment (PromoFaux) [reply test/test_any_utils.py:10]: Ah. Then - we are ok.

The original logic of the test was to prove that adding something that already existed would not duplicate. Maybe we can create a separate test to make this clearer?

Comment (yubiuser) [reply test/test_any_utils.py:10]: This is a good idea. 
We have separted the functions and as you wrote in `utils.sh` ;-)
> New functions must have a test added for them in test/test_any_utils.py

**Review by yubiuser — APPROVED** [review]

**Pull request merged**